hadoop-hdds/common/src/main/java/org/apache/hadoop/utils/db/TypedTable.java

```diff
@@ -79,6 +79,12 @@ public boolean isEmpty() throws IOException {
     return rawTable.isEmpty();
   }
 
+  @Override
+  public boolean isExist(KEY key) throws IOException {
+    return cache.get(new CacheKey<>(key)) != null ||
```

Comment:
Here, we need to check value also along with key.

A case like createBucket, DeleteBucket, we will have the key in the cache for that bucket, but with the value null.

Reply:
@bharatviswa504 Yes, the isExist method will be called with the lock. 

I am not sure I understand the createBucket, Delete Bucket scenario you have mentioned. In the case of create bucket, if the cache is used why would the value be null? And even if the value is null, we fall back to actual DB.  In the case of Delete bucket, shouldn't the value of get(key) be null? And it wont exist in the DB also?

Reply:
yes, think of a case like where flush to om DB has not happened, it is only in the cache. so for key exist we need to check the value also in the cache, as for delete case we set value as Optional.absent().
https://github.com/apache/hadoop/blob/trunk/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/OMBucketDeleteRequest.java#L103

Can you add a java doc to the method caller should acquire/synchronize before calling this method.